openqa-label-known-issues

```diff
@@ -143,7 +136,17 @@ investigate_issue() {
     echo "$reason" >> "$out"
     if [[ "$curl_output" != "200" ]] && [[ "$curl_output" != "301" ]]; then
         # if we can not even access the page it is something more critical
-        handle_unreachable_or_no_log "$id" "$testurl" "$out"
+        handle_unreachable "$testurl" "$out"
```

Comment:
`handle_unreachable` does not always exit, so we would continue even if the url is unreachable? Maybe you can influence that with a `return ...` and check the return value?

Reply:
Looking. I considered this but the examples I run didnt fall into this case.

Reply:
First the function didnt have any exit even before the refactoring. 
But where do you want it to exit. `handle_unreachable` looks to me more like a diagnostic operation. it should continue unless...?

Reply:
it didn't need to exit before, because it was in a branch that did not do anything after that

Reply:
But if @okurz is ok with it, then ignore me ;-)
He wrote the original function.

Reply:
ok. but not sure I follow. 

when I stop vpn and try to access url I get 
```❯ dry_run=1 host=openqa.suse.de ./openqa-label-known-issues https://openqa.suse.de/tests/14890572
Requesting jobs/14890572 via openqa-cli
Can't connect: Name or service not known
./openqa-label-known-issues: ERROR: line 124
./openqa-label-known-issues: ERROR: line 124
```
So does this even used if the url is unreachable anyway?

Reply:
As I tried to explain when we talked about it in Jitsi you can't reach `handle_unreachable` if the server is not reachable at all. The `handle_unreachable` tries to handle an intermittent outage or other problems. I guess we can accept the PR as is even if there is a slight unexpected change of behaviour in an execution branch which in practice might not be reachable anymore.

Reply:
But i still need to understand where is it expected to exit? I am gonna check again on weekend

Reply:
Not an expert but the `-e` makes the script behaves a little different and the return code from the `handle_unreachable` is not return as expected. I used also exits instead of the returns but the script output shows some errors, as it seems to execute something additional after the last statement. You can see this by adding `-xv` to the script.
So if @perlpunk is ok with you, can we proceed and accept the current changes and maybe create another ticket to refactor/review the structure of the script later?